src/pass/verify_memory.cc

```diff
@@ -0,0 +1,187 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file verify_memory.cc
+ * \brief Pass to check if memory access locations are consistent
+ * in terms of device type.
+ */
+#include <tvm/ir.h>
+#include <tvm/ir_mutator.h>
+#include <tvm/ir_pass.h>
+#include <tvm/operation.h>
+
+namespace tvm {
+namespace ir {
+namespace {
```

Comment:
This is great, for our case, actually we know the device type from build module, so instead what we can do is directly pass in the device type as second argument of VerifyMemoryAccess, and do static checking at compile time

Reply:
This can be done in build function(instead of lower) where the device type is known

Reply:
Hi Tianqi,

Thanks for your advice!

I made an update as per your suggestions. Now I am taking advantage of dev_type variable directly available in build(), and use IRVisitor instead of IRMutator. Such verification is performed statically.